clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/provider/agent/CloudFoundryCachingAgent.java

```diff
@@ -16,50 +16,152 @@
 
 package com.netflix.spinnaker.clouddriver.cloudfoundry.provider.agent;
 
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.MapperFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.netflix.spinnaker.cats.agent.*;
+import com.netflix.spinnaker.cats.cache.CacheData;
 import com.netflix.spinnaker.cats.provider.ProviderCache;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.cache.Keys;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.cache.ResourceCacheData;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.client.CloudFoundryClient;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.model.CloudFoundryApplication;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.model.CloudFoundryLoadBalancer;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.model.Views;
 import com.netflix.spinnaker.clouddriver.cloudfoundry.provider.CloudFoundryProvider;
-import com.netflix.spinnaker.clouddriver.cloudfoundry.security.CloudFoundryCredentials;
-import com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace;
 import lombok.Getter;
-import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
+import java.util.*;
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE;
+import static com.netflix.spinnaker.clouddriver.cloudfoundry.cache.Keys.Namespace.*;
+import static java.util.Collections.emptyMap;
+import static java.util.stream.Collectors.toSet;
 
-@RequiredArgsConstructor
 @Getter
 @Slf4j
 public class CloudFoundryCachingAgent implements CachingAgent, AccountAware {
```

Comment:
I suspect that having a single caching agent for all data types will become a problem for larger CF deployments, but it works for an initial go around.

Reply:
Yep. We thought about it. I suppose what we know is that no CF deployment contains enough apps to be a problem. If it were, I'd retire and let somebody else figure this out.